charts/kyverno/templates/_helpers.tpl

```diff
@@ -116,3 +116,26 @@ maxUnavailable: {{ .Values.podDisruptionBudget.maxUnavailable }}
 {{- define "kyverno.imagePullSecret" }}
 {{- printf "{\"auths\":{\"%s\":{\"auth\":\"%s\"}}}" .registry (printf "%s:%s" .username .password | b64enc) | b64enc }}
 {{- end }}
+
+
+{{- define "kyverno.resourceFilters" -}}
+{{- $resourceFilters := .Values.config.resourceFilters }}
+{{- if .Values.excludeKyvernoNamespace }}
+  {{- $resourceFilters = prepend .Values.config.resourceFilters (printf "[*,%s,*]" (include "kyverno.namespace" .)) }}
+{{- end }}
+{{- tpl (join "" $resourceFilters) . }}
+{{- end }}
+
+{{- define "kyverno.webhooks" -}}
+{{- $excludeDefault := dict "key" "kubernetes.io/metadata.name" "operator" "NotIn" "values" (list (include "kyverno.namespace" .)) }}
+{{- $newWebhook := list }}
+{{- range $webhook := .Values.config.webhooks }}
+  {{- $namespaceSelector := default dict $webhook.namespaceSelector }}
+  {{- $matchExpressions := default list $namespaceSelector.matchExpressions }}
+  {{- if $.Values.excludeKyvernoNamespace }}
+    {{- $namespaceSelector = dict "matchLabels" $namespaceSelector.matchLabels "matchExpressions" (append $matchExpressions $excludeDefault) }}
+  {{- end }}
+  {{- $newWebhook = merge (omit $webhook "namespaceSelector") (dict "namespaceSelector" $namespaceSelector) }}
```

Comment:
Should be an append here, we expect a list

Reply:
Fixed

Reply:
Actually, using `append` results in Helm render issues, merge does work.  Using one of the CI files get this:

```
  webhooks: "{\"namespaceSelector\":{\"matchExpressions\":[{\"key\":\"kubernetes.io/metadata.name\",\"operator\":\"NotIn\",\"values\":[\"kube-system\",\"kyverno\"]},{\"key\":\"kubernetes.io/metadata.name\",\"operator\":\"NotIn\",\"values\":[\"default\"]}],\"matchLabels\":null}}"

```

Reply:
It can't be an object, it has to be a list else the controller won't be able to process it.

Reply:
I see the issue, trying to fix.

Reply:
I fixed, so now get this:

```
  webhooks: "[{\"namespaceSelector\":{\"matchExpressions\":[{\"key\":\"kubernetes.io/metadata.name\",\"operator\":\"NotIn\",\"values\":[\"kube-system\",\"kyverno\"]},{\"key\":\"kubernetes.io/metadata.name\",\"operator\":\"NotIn\",\"values\":[\"default\"]}],\"matchLabels\":null}}]"
```

Reply:
Does this work ?
```
{{- define "kyverno.webhooks" -}}
{{- $webhooks := .Values.config.webhooks }}
{{- if .Values.excludeKyvernoNamespace }}
  {{- $excludeDefault := dict "key" "kubernetes.io/metadata.name" "operator" "NotIn" "values" (list (include "kyverno.namespace" .)) }}
  {{- $webhooks = list }}
  {{- range .Values.config.webhooks }}
    {{- $namespaceSelector := default dict .namespaceSelector }}
    {{- $matchExpressions := append (default list $namespaceSelector.matchExpressions) $excludeDefault }}
    {{- $namespaceSelector = dict "matchLabels" $namespaceSelector.matchLabels "matchExpressions" $matchExpressions }}
    {{- $webhooks = append $webhooks (merge (omit . "namespaceSelector") (dict "namespaceSelector" $namespaceSelector)) }}
  {{- end }}
{{- end }}
{{- $webhooks | toJson }}
{{- end }}
```

Reply:
So that would except if you don't check if `omit . "namespace"` selector is empty, you end up with empty dict in the list, which isn't likely a problem just a bit unnecessary so just adding extra condition on that.

Reply:
Given the merge statement `merge (omit . "namespaceSelector") (dict "namespaceSelector" $namespaceSelector)` we should not end up with an empty dict right ?

Reply:
Yep, the merge does solve that, I pushed changed to address this.

Reply:
:+1: